host/versioning_test.go

```diff
@@ -0,0 +1,249 @@
+	}
+	s.sdkClient = sdkClient
+}
+
+func (s *versioningIntegSuite) TearDownTest() {
+	s.sdkClient.Close()
+}
+
+func TestVersioningIntegrationSuite(t *testing.T) {
+	flag.Parse()
+	suite.Run(t, new(versioningIntegSuite))
+}
+
+func (s *versioningIntegSuite) TestBasicVersionUpdate() {
+	ctx := NewContext()
+	tq := "integration-versioning-basic"
+	s.prepareQueue(ctx, tq)
+
+	res, err := s.engine.UpdateWorkerBuildIdOrdering(ctx, &workflowservice.UpdateWorkerBuildIdOrderingRequest{
+		Namespace:          s.namespace,
+		TaskQueue:          tq,
+		VersionId:          &taskqueuepb.VersionId{WorkerBuildId: "foo"},
+		PreviousCompatible: nil,
+		BecomeDefault:      true,
+	})
+	s.NoError(err)
+	s.NotNil(res)
+
+	res2, err := s.engine.GetWorkerBuildIdOrdering(ctx, &workflowservice.GetWorkerBuildIdOrderingRequest{
+		Namespace: s.namespace,
+		TaskQueue: tq,
+	})
+	s.NoError(err)
+	s.NotNil(res2)
+	s.Equal("foo", res2.CurrentDefault.GetVersion().GetWorkerBuildId())
+}
+
+func (s *versioningIntegSuite) TestSeriesOfUpdates() {
+	ctx := NewContext()
+	tq := "integration-versioning-series"
+	s.prepareQueue(ctx, tq)
+
+	for i := 0; i < 10; i++ {
+		res, err := s.engine.UpdateWorkerBuildIdOrdering(ctx, &workflowservice.UpdateWorkerBuildIdOrderingRequest{
+			Namespace:          s.namespace,
+			TaskQueue:          tq,
+			VersionId:          &taskqueuepb.VersionId{WorkerBuildId: fmt.Sprintf("foo-%d", i)},
+			PreviousCompatible: nil,
+			BecomeDefault:      true,
+		})
+		s.NoError(err)
+		s.NotNil(res)
+	}
+	res, err := s.engine.UpdateWorkerBuildIdOrdering(ctx, &workflowservice.UpdateWorkerBuildIdOrderingRequest{
+		Namespace:          s.namespace,
+		TaskQueue:          tq,
+		VersionId:          &taskqueuepb.VersionId{WorkerBuildId: "foo-2.1"},
+		PreviousCompatible: &taskqueuepb.VersionId{WorkerBuildId: "foo-2"},
+		BecomeDefault:      false,
+	})
+	s.NoError(err)
+	s.NotNil(res)
+
+	res2, err := s.engine.GetWorkerBuildIdOrdering(ctx, &workflowservice.GetWorkerBuildIdOrderingRequest{
+		Namespace: s.namespace,
+		TaskQueue: tq,
+	})
+	s.NoError(err)
+	s.NotNil(res2)
+	s.Equal("foo-9", res2.CurrentDefault.GetVersion().GetWorkerBuildId())
+	s.Equal(1, len(res2.CompatibleLeaves))
+	s.Equal("foo-2.1", res2.CompatibleLeaves[0].GetVersion().GetWorkerBuildId())
+}
+
+func (s *versioningIntegSuite) TestLinkToNonexistentCompatibleVersionReturnsNotFound() {
+	ctx := NewContext()
+	tq := "integration-versioning-compat-not-found"
+	s.prepareQueue(ctx, tq)
+
+	res, err := s.engine.UpdateWorkerBuildIdOrdering(ctx, &workflowservice.UpdateWorkerBuildIdOrderingRequest{
+		Namespace:          s.namespace,
+		TaskQueue:          tq,
+		VersionId:          &taskqueuepb.VersionId{WorkerBuildId: "foo"},
+		PreviousCompatible: &taskqueuepb.VersionId{WorkerBuildId: "i don't exist yo"},
+	})
+	s.Error(err)
+	s.Nil(res)
+	s.IsType(&serviceerror.NotFound{}, err)
+}
+
+func (s *versioningIntegSuite) TestVersioningStateNotDestroyedByOtherUpdates() {
```

Comment:
Can you please add some clarification what it is exactly that you're testing here?